Orange/data/pandas_compat.py

```diff
@@ -203,12 +216,16 @@ def vars_from_df(df, role=None, force_nominal=False):
                                    discrete.categories.astype(str).tolist())
             attrs.append(var)
             Xcols.append(column)
-            Xexpr.append(lambda s, _: np.asarray(
-                s.astype('category').cat.codes.replace(-1, np.nan)
-            ))
+
+            def to_cat(s, _):
+                x = s.astype("category").cat.codes
+                # it is same than x.replace(-1, np.nan), but much faster
+                x = x.where(x != -1, np.nan)
+                return np.asarray(x)
+
+            Xexpr.append(to_cat)
         elif _is_datetime(s):
             var = TimeVariable(str(column))
-            s = pd.to_datetime(s, infer_datetime_format=True)
```

Comment:
I think this line dos do not have an effect? @irgolic is it correct or I missed something?

Reply:
I think this was a remnant of @kernc's table to dataframe implementation; in lieu of backwards compatibility, I tried to keep the behavior as similar as possible